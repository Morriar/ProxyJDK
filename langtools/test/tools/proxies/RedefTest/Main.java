/*
 * @test
 * @summary redef
 * @author Alexandre Terrasa
 *
 * @compile RealSubject/RunnableRealSubject.java Main.java
 */

package TMPProxyTests.RedefTest;

import java.io.Serializable;
import TMPProxyTests.RedefTest.RealSubject.RunnableRealSubject;

public class Main extends proxy java.lang.Runnable implements java.lang.Runnable {
	
	public static void main(String[] args) {
		
		Main prxy = new Main(new RunnableRealSubject("RedefTest"));
		prxy.run();
		prxy.addedMethod();
	}

	public Main(Runnable realSubject) {
		super(realSubject);
	}
	
	public void run() {
		super.run();
		System.out.println("-> This is a redef <-");
	}
	
	public void addedMethod() {
		System.out.println("added");
	}
	
}
