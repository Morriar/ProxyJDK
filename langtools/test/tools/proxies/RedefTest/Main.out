run by RedefTest
-> This is a redef <-
added
